### PR TITLE
from_slice

### PR DESCRIPTION
**Pull request by FintanH**
Add `from_slice` to allow callers to get a `NonEmpty` from a `&[T]` (maybe)

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -111,6 +111,37 @@ impl<T> NonEmpty<T> {
...
+    ///
+    /// let non_empty_vec = NonEmpty::from_vec(vec!([1, 2, 3, 4, 5]));
+    /// assert!(non_empty_vec.is_some());
+    ///
+    /// let empty_vec: Option<NonEmpty<u32>> = NonEmpty::from_vec(Vec::new());
+    /// assert!(empty_vec.is_none());
+    /// ```
+    pub fn from_vec(vec: Vec<T>) -> Option<NonEmpty<T>> {
+        let mut vec = vec;
+        let head = vec.pop();
```
Comment (cloudhead) [inline src/lib.rs]: This will pop the *last* element of the vec, I don't think that's what you want?

File: src/lib.rs [hunk 2]
```diff
@@ -122,21 +122,18 @@ impl<T> NonEmpty<T> {
...
-                let mut result = NonEmpty::new(t);
-                for u in vec {
-                    result.push(u)
-                }
+    pub fn from_vec(vec: &[T]) -> Option<NonEmpty<&T>> {
+        let split = vec.split_first();
+        match split {
+            Some((h, t)) => {
+                let mut result: NonEmpty<&T> = NonEmpty::new(h);
+                t.iter().for_each(|u| result.push(u));
```
Comment (cloudhead) [inline src/lib.rs]: Instead of this, you could just do `NonEmpty(h, t)`

Comment (cloudhead) [reply src/lib.rs]: Well, you may have to clone the `t`, since it returns a slice.

Comment (FintanH) [reply src/lib.rs]: Would it not be better to have refs and let the caller decide?

**Comment by cloudhead** [comment]
This is now merged, thanks!

**Pull request closed without merging**